Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm

```diff
@@ -1211,11 +1211,21 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
 
 void WebPageProxy::compositionSessionDidReceiveTextWithReplacementRange(const WebCore::WritingTools::Session& session, const WebCore::AttributedString& attributedText, const WebCore::CharacterRange& range, const WebCore::WritingTools::Context& context, bool finished)
 {
+    internals().writingToolsTextReplacementsFinished = finished;
+    internals().partialIntelligenceTextPonderingAnimationCount += 1;
+
     legacyMainFrameProcess().send(Messages::WebPage::CompositionSessionDidReceiveTextWithReplacementRange(session, attributedText, range, context, finished), webPageIDInMainFrameProcess());
 }
 
 void WebPageProxy::writingToolsSessionDidReceiveAction(const WebCore::WritingTools::Session& session, WebCore::WritingTools::Action action)
 {
+    if (action == WebCore::WritingTools::Action::Restart && session.compositionType != WebCore::WritingTools::Session::CompositionType::SmartReply) {
```

Comment:
Why is smart reply special cased here? Wouldn't `writingToolsTextReplacementsFinished` be in an incorrect state.